### DISCUSSION TITLE
How to confirm before closing the window?

### DISCUSSION THREAD

**Discussion by photino** [DioxusLabs/dioxus#2988] [Q&A]
In a desktop application built with dioxus, the user should confirm the action before closing the window? How can we listen to window event? Thanks!

**Comment by ealmloff** [comment]
You can listen to the window's events with https://docs.rs/dioxus-desktop/0.6.0-alpha.2/dioxus_desktop/fn.use_wry_event_handler.html, but dioxus currently doesn't expose a way to prevent the default behavior

**Reply by photino** [reply]
Thanks! It is a common behavior to alert the user to save a file when closing the window in a desktop application. We hope that dioxus will support it in a future version.
